src/components/adaptive-cards/AdaptiveCard/AdaptiveCard.jsx

```diff
@@ -95,21 +99,57 @@ export default function AdaptiveCard({
       };
     });
   };
+
   const setInput = useCallback((input) => {
     setInputs((prevInputs) => ({...prevInputs, [input.id]: input}));
   }, [setInputs]);
+
   const getValue = (id, defval = '') => ((id in inputs && inputs[id].value) ? inputs[id].value : defval);
-  const getAllValues = () => Object.entries(inputs).reduce((allValues, [id, input]) => (
-    {...allValues, [id]: input.value}
-  ), {});
+
+  const getAllValues = () => Object.entries(inputs).reduce(
+    (allValues, [id, input]) => (
+      {...allValues, [id]: input.value}
+    ), {},
+  );
+
+  const getError = (id, defval = undefined) => (
+    (id in inputs && inputs[id].error) ? inputs[id].error : defval
+  );
+
+  const validate = () => {
+    let valid = true;
+
+    Object.entries(inputs).forEach(([, input]) => {
+      let error;
+
+      if (input.isRequired && !input.value && input.value !== 0) {
+        error = input.errorMessage;
+        valid = valid && !error;
```

Comment:
the updating of the `valid` flag can be done only once, after the last `else if`, together with `setInput`